browser/ui/webui/brave_new_tab_message_handler.cc

```diff
@@ -406,6 +412,8 @@ void BraveNewTabMessageHandler::HandleSaveNewTabPagePref(
     settingsKey = kNewTabPageShowClock;
   } else if (settingsKeyInput == "showStats") {
     settingsKey = kNewTabPageShowStats;
+  } else if (settingsKeyInput == "showToday") {
```

Comment:
I know this is just adding; but it might be good to have a proper map here. We can do something like:
```cpp
#include "base/containers/flat_map.h"
#include "base/no_destructor.h"

//..

void BraveNewTabMessageHandler::HandleSaveNewTabPagePref(
    const base::ListValue* args) {

    //..

    static const base::NoDestructor<base::flat_map<const char*, const char*>> bool_mappings({
        {"showBackgroundImage", kNewTabPageShowBackgroundImage},
        {"brandedWallpaperOptIn", kNewTabPageShowSponsoredImagesBackgroundImage},
        //..
        {"showToday", kNewTabPageShowToday},
    });

    auto it = bool_mappings->find(settingsKeyInput);
    if (it == bool_mappings->end()) {
        LOG(ERROR) << "Invalid setting key: " << settingsKeyInput;
        return;
    }
    prefs->SetBoolean(it->second, settingsValueBool);
}
```

Reply:
I don't think I'll do that in this PR, there's too much going on here. Also I think there is an argument for removing this and using the `chrome.settingsPrivate` API.

Reply:
OK SGTM 👍